packages/core-modules/contracts/modules/CoreElectionModule.sol

```diff
@@ -133,37 +139,46 @@ contract CoreElectionModule is IElectionModule, ElectionStorage, OwnableMixin {
         _electionStore().nextNominationPeriodPercent = percent;
     }
 
-    function elect(address[] memory candidates) external view override {
-        uint seatCount = _electionStore().nextSeatCount;
+    function elect(address[] memory candidates) external override {
+        ElectionStore storage store = _electionStore();
+        ElectionData storage electionData = store.electionData;
+
+        uint seatCount = store.nextSeatCount;
 
-        if (candidates.length != seatCount) {
+        if (candidates.length == 0 || candidates.length > seatCount) {
             revert InvalidCandidatesCount();
         }
 
-        // TODO: if msg.sender already voted, rollback previous votes.
+        if (electionData.addressVoted[msg.sender]) {
+            // TODO: if msg.sender already voted, rollback previous votes and allow to re-vote
+            revert AlreadyVoted();
+        }
 
-        // TODO: Assign votes to each address (same to all)
-        // uint votePower = ERC20(_electionStore().electionTokenAddress).balanceOf(msg.sender);
+        // The voting power is the amount of votes we are going to assign to the given candidates
+        uint votePower = ERC20(store.electionTokenAddress).balanceOf(msg.sender);
 
-        // TODO: Recalculate top [seatsCount] nominees
-        //   _minimunIdx;
-        //   _minimumValue;
-        //   if votes > minimum
-        //     electionTopNominees[last][minumunNomineeTop] = msg.sender
-        //     for () // get new minimum (idx & value)
+        for (uint i = 0; i < candidates.length; i++) {
```

Comment:
we should store (somewhere) the original list of candidates voted in the ballot

Reply:
If instead of assigning the same amount of voting power to all the candidates we divide it equally, we could remove this check because the repeated candidates wouldn't affect the final result.

Reply:
I will store the list of candidates, vote author, and voting power assigned.